.github/workflows/deploy-pr-preview.yaml

```diff
@@ -155,6 +156,35 @@ jobs:
           APP_PROPS_BASE64: ${{ secrets.CARBON_APP_PROPS_BASE64 }}
         continue-on-error: true
 
+      - name: Build GREENHOUSE if changes detected
+        if: github.event.action != 'closed' && contains(needs.run-detect-changes.outputs.changes, 'greenhouse')
+        run: |
+          # collect the necessary information
+          entry_file=$(jq -r '.module // .main' $PACKAGE_PATH/package.json)
+          build_folder=$(dirname $entry_file)
+          package_name=$(jq -r '.name' $PACKAGE_PATH/package.json)
+
+          # Run build using turbo
+          npx turbo run build --filter $package_name
+
+          # Copy the necessary files
+          mkdir -p "$DEPLOY_PATH/$TARGET_FOLDER"
+          cp -r "$PACKAGE_PATH/$build_folder" "$DEPLOY_PATH/$TARGET_FOLDER/$build_folder"
+          cp "$PACKAGE_PATH/public/favicon.ico" "$DEPLOY_PATH/$TARGET_FOLDER/"
+          cp "$PACKAGE_PATH/public/favicon.svg" "$DEPLOY_PATH/$TARGET_FOLDER/"
+
+          # Generate index.html and appProps.json
+          cp "$PACKAGE_PATH/docker/index.html" "$DEPLOY_PATH/$TARGET_FOLDER/index.html"
```

Comment:
that will break once we moved to vite. Just keep it in mind

Reply:
Yes, I’m aware, thanks! The corresponding changes will be included in the Vite migration PR.